client/lib/user/user.ts

```diff
@@ -0,0 +1,67 @@
+/**
+ * External Dependencies
+ */
+import { EventEmitter } from 'events';
+import { Language } from '../../languages';
+import { GravatarOptions } from './gravatar';
+import { URL } from '../../types';
+
+/**
+ * Takes an optional callback function. If no callback is passed in, it will return a promise.
+ **/
+export interface RequestHandler {
+	( fn: ( err: unknown, res: unknown, headers: unknown ) => unknown ): unknown;
+	(): Promise< unknown >;
+}
+
+export interface User extends EventEmitter {
+	initialize: () => Promise< void >;
+	clearStoreIfChanged: () => void;
+	get: () => UserData;
+	fetch: () => Promise< void >;
+	handleFetchFailure: ( error: Error ) => void;
+	handleFetchSuccess: ( userdata: UserData ) => void;
+	getLanguage: () => Language | void;
+	getAvatarUrl: ( options: GravatarOptions ) => URL;
+	clear: () => Promise< void > | void;
+	sendVerificationEmail: RequestHandler;
+	set: ( attributes: UserData ) => boolean;
+	decrementSiteCount: () => void;
+	incrementSiteCount: () => boolean | void;
```

Comment:
cc: @jsnajdr I don't think the purpose of `incrementSiteCount()` is to return the boolean from `this.set()`

as far as the code looks to me I think that's one of those early-aborts I mentioned earlier. that is, I think that…

```ts
return this.set(…);
```

…might have well have been written as this…

```ts
this.set(…);
return;
```

…but we typically use the short-hand with `return` because it's JavaScript 🙃 

if it makes the types more consistent and obvious and we need to rewrite that `return` statement to make this `() => void` then I'm all for it.

Reply:
(I did check and it only appears like this is called once and without relying on the return value)

Reply:
hmm yea that is bugging me too, I'd like to update the type to `() => void` and leave the JS code as is, just so I can say there's no runtime changes and keep testing simple.

Reply:
> I'd like to update the type to () => void and leave the JS code as is, just so I can say there's no runtime changes and keep testing simple.

that sounds like a fair thing to do, especially if we are making a declarations file intending to show the usable API